lib/index.ts

```diff
@@ -7,24 +7,39 @@ import { Docker, DockerOptions } from './docker';
 import { buildResponse } from './response-builder';
 
 export {
+  EnvDetails,
   inspect, dockerFile,
 };
 
+// Additional details which can help the analyzer
+// make a decision
+//
+interface EnvDetails {
+  dockerFile?: dockerFile.DockerFileAnalysis;
+}
+
 function inspect(root: string, targetFile?: string, options?: any) {
   const dockerOptions = options ? {
     host: options.host,
     tlsverify: options.tlsverify,
     tlscert: options.tlscert,
-    tlscacert: options.tlscacert,
+     tlscacert: options.tlscacert,
     tlskey: options.tlskey,
   } : {};
   const targetImage = root;
-  return Promise.all([
-    getRuntime(dockerOptions),
-    getDependencies(targetImage, dockerOptions),
-    dockerFile.readDockerfileAndAnalyse(targetFile),
-  ])
-    .then((res) => buildResponse(res[0], res[1], res[2], options));
+
+  return dockerFile.readDockerfileAndAnalyse(targetFile)
+    .then(dockerFileResult => {
```

Comment:
Here we first resolve the dockerfile promise and then do the runtime and dependencies.